radio@hslbck.gmail.com/addChannelDialog.js

```diff
@@ -156,43 +157,48 @@ var AddChannelDialog = new Lang.Class({
     },
 
     _closeDialog: function() {
+        this.close();
         if (oldChannel != null) {
-            this.close();
             this.channelListDialog = new ChannelListDialog.ChannelListDialog();
             this.channelListDialog.open();
         }
-        else {
-            this.close();
-        }
     }
 });
 
 // get the valid stream address
 function getStreamAddress(input) {
-    let streamAddress;
     let regexp = /\.(m3u|m3u8|pls)/i;
 
     // test for m3u / pls
     if (input.search(regexp) != -1) {
 
         // get file
         var message = Soup.Message.new('GET', input);
-        _httpSession.send_message(message);
-        // request ok
-        if (message.status_code === 200) {
-            let content = message.response_body.data;
-            let contentLines = content.split('\n');
-            // look for stream url
-            for (var line in contentLines) {
-                if (contentLines[line].search(/http:/i) != -1) {
-                    // get url
-                    streamAddress = contentLines[line].slice((contentLines[line].search(/http:/))) //, contentLines[line].search(/\n/));
-                    break;  // break on the first occurrence
-                }
-            }
-        }
-    } else {
-        streamAddress = input; // case for valid stream address
+
+        if (message != null) {
+          _httpSession.send_message(message);
+
+          // request ok
+          if (message.status_code === 200) {
+              let content = message.response_body.data;
+              let contentLines = content.split('\n');
+              // look for stream url
+              for (var line in contentLines) {
+                  if (contentLines[line].search(/http:/i) != -1) {
+                      // get url
+                      return contentLines[line].slice((contentLines[line].search(/http:/))) //, contentLines[line].search(/\n/));
+                  }
+              }
+          }
+
+          this._addMessage(_("status code different from 200 or data in loop not set properly"));
+
+      } else {
+        this._addMessage(_("please check input"));
```

Comment:
@hslbck, so, looking at the logs, I saw some errors when `message` was `null`. I added a test to prevent this.

So now, there's two different errors. I'm not familiar with the gnome-shell API and find the doc pretty unclear. In this case, what would be a correct equivalent to the non-existing `this._addMessage` function ?

Moreover, current text should be changed in both cases, but I'm not sure what we should tell to the user, any idea ?

Also, does the `return;` (see below) is OK ?

Reply:
`_addMessage` was introduced by me specific for the `searchDialog.js`. It won't work in `addChannelDialog.js`. There are 2 options here, first is to add a new label below the input and set the text to the error message. Second when you hit `alt-f2` and enter `r` on Wayland a nice error message appears. I haven't checked the gnome shell source code, but having this here would be nice.

The function could be exited with `return;` but adding the channel to the list should not be possible and the popup should still be visible with the error text.

Reply:
I started looking at the gnome-shell source code, I found [this](https://bitbucket.org/mathematicalcoffee/gnome-shell-doc/src/bae8d9bd9403747ee6ec8d95e702a8b35f74009d/js/ui/runDialog.js?at=gnome-3-2&fileviewer=file-view-default#runDialog.js-376), now I'm trying to find the `global.reexec_self();` source code...